src/prefect/runner/runner.py

```diff
@@ -1047,17 +1047,32 @@ async def _submit_run_and_capture_errors(
     ) -> Union[Optional[int], Exception]:
         run_logger = self._get_flow_run_logger(flow_run)
 
-        if flow_run.deployment_id:
-            deployment = await self._client.read_deployment(flow_run.deployment_id)
-        if deployment and deployment.global_concurrency_limit:
-            limit_name = deployment.global_concurrency_limit.name
-            concurrency_ctx = concurrency
-        else:
-            limit_name = ""
-            concurrency_ctx = asyncnullcontext
+        disable_concurrency = (
+            os.environ.get("PREFECT__DISABLE_CONCURRENCY_SLOT_ACQUISITION", "").lower()
+            == "true"
+        )
 
         try:
-            async with concurrency_ctx(limit_name, max_retries=0, strict=True):
+            if not disable_concurrency and flow_run.deployment_id:
+                deployment = await self._client.read_deployment(flow_run.deployment_id)
+                if deployment and deployment.global_concurrency_limit:
+                    async with concurrency(
+                        deployment.global_concurrency_limit.name,
+                        max_retries=0,
+                        strict=True,
+                    ):
+                        status_code = await self._run_process(
+                            flow_run=flow_run,
+                            task_status=task_status,
+                            entrypoint=entrypoint,
+                        )
+                else:
+                    status_code = await self._run_process(
+                        flow_run=flow_run,
+                        task_status=task_status,
+                        entrypoint=entrypoint,
+                    )
```

Comment:
Could you conditionally set up a `concurrency` or an `asyncnullcontext` so we don't need to call `self._run_process` in three different conditional branches? We have an existing `asyncnullcontext` implementation in `src/prefect/utilities/asyncutils.py`.

Reply:
Yeah, totally